Servers/UI/OJS.Servers.Ui/ClientApp/src/common/urls/compose-client-urls.ts

```diff
@@ -0,0 +1,30 @@
+import { CONTESTS_PATH } from './client-urls';
+
+const composeParticipationTypeResultsFullRoute =
+    (id: number, participationType: string) => `/${CONTESTS_PATH}/${id}/${participationType}/results/full`;
+
+const composeContestsWithSelectedCategoryAndStrategyUrl =
+    (categoryId?: number, strategyId?: number) => `/contests?category=${categoryId}&strategy=${strategyId}`;
+
+const composeContestBreadcrumbUrl = (breadcrumbItemId: number) => `/contests?category=${breadcrumbItemId}`;
```

Comment:
These are very similar. `categoryId` is the same as `breadcrumbItemId`?
You can unify in a single function with optional parameters.

Reply:
they are, but woudlnt that be semantically wrong?

Reply:
Why so? Semantically they are used for the same thing right?

Reply:
makes sense, pushed fix